pymumble_py3/callbacks.py

```diff
@@ -1,91 +1,102 @@
 import threading
+import typing
+
+from .constants import *
+from .errors import UnknownCallbackError
+
+_Callback = typing.Callable[..., typing.Any]
+_Callbacks = typing.List[_Callback]
 
 
-class CallBacks(dict):
+class CallBacks(typing.Dict[str, typing.Optional[_Callbacks]]):
     """
     Define the callbacks that can be registered by the application.
     Multiple functions can be assigned to a callback using "add_callback"
 
     The call is done from within the pymumble loop thread, it's important to
     keep processing short to avoid delays on audio transmission
     """
-    def __init__(self):
-        self.update({
-            PYMUMBLE_CLBK_CONNECTED: None,  # Connection succeeded
-            PYMUMBLE_CLBK_CHANNELCREATED: None,  # send the created channel object as parameter
-            PYMUMBLE_CLBK_CHANNELUPDATED: None,  # send the updated channel object and a dict with all the modified fields as parameter
-            PYMUMBLE_CLBK_CHANNELREMOVED: None,  # send the removed channel object as parameter
-            PYMUMBLE_CLBK_USERCREATED: None,  # send the added user object as parameter
-            PYMUMBLE_CLBK_USERUPDATED: None,  # send the updated user object and a dict with all the modified fields as parameter
-            PYMUMBLE_CLBK_USERREMOVED: None,  # send the removed user object and the mumble message as parameter
-            PYMUMBLE_CLBK_SOUNDRECEIVED: None,  # send the user object that received the sound and the SoundChunk object itself
-            PYMUMBLE_CLBK_TEXTMESSAGERECEIVED: None,  # Send the received message
-            PYMUMBLE_CLBK_CONTEXTACTIONRECEIVED: None,  # Send the contextaction message
-        })
-            
-    def set_callback(self, callback, dest):
+
+    def __init__(self) -> None:
+        self.update(
+            {
+                PYMUMBLE_CLBK_CONNECTED: None,  # Connection succeeded
+                PYMUMBLE_CLBK_CHANNELCREATED: None,  # send the created channel object as parameter
+                PYMUMBLE_CLBK_CHANNELUPDATED: None,  # send the updated channel object and a dict with all the modified fields as parameter
+                PYMUMBLE_CLBK_CHANNELREMOVED: None,  # send the removed channel object as parameter
+                PYMUMBLE_CLBK_USERCREATED: None,  # send the added user object as parameter
+                PYMUMBLE_CLBK_USERUPDATED: None,  # send the updated user object and a dict with all the modified fields as parameter
+                PYMUMBLE_CLBK_USERREMOVED: None,  # send the removed user object and the mumble message as parameter
+                PYMUMBLE_CLBK_SOUNDRECEIVED: None,  # send the user object that received the sound and the SoundChunk object itself
+                PYMUMBLE_CLBK_TEXTMESSAGERECEIVED: None,  # Send the received message
+                PYMUMBLE_CLBK_CONTEXTACTIONRECEIVED: None,  # Send the contextaction message
+            }
+        )
+
+    def set_callback(self, callback: str, dest: _Callback) -> None:
```

Comment:
I feel like this change is a rather subjective one. It just makes the lines longer because the tool decided you need to have this hierarchy.

Reply:
yup, I completely agree with you. I would never format it like this manually.